src/routes/[topicSlug]/[tableSlug]/[categorySlug].svelte

```diff
@@ -83,17 +85,16 @@
 
   const initialisePage = () => {
     category = getCategoryBySlug(tableSlug, categorySlug);
-    table = category ? tables[category.table] : null;
-    populatesSelectedData(table.name, table.categoriesArray, category.code);
-    fetchCensusData(new LegacyCensusDataService(), category.code, null);
+    table = category ? filterSelectedTable(metadata, category) : null;
+    populatesSelectedData(table.name, table.categories, category.code, table.totalCode);
```

Comment:
OOS for this PR, but this could use some comments to explain whats going on, at the moment its a bit of a shopping list

Reply:
yep agreed.  some of this would be simplified by the remaining bits of metadata refactoring that need to happen